components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -5343,6 +5344,9 @@ public int addAPI(API api, int tenantId, String organization) throws APIManageme
                     .getTenantAwareUsername(APIUtil.replaceEmailDomainBack(api.getId().getProviderName()));
             recordAPILifeCycleEvent(apiId, null, APIStatus.CREATED.toString(), tenantUserName, tenantId,
                     connection);
+            // persist throttle Limit
```

Comment:
add a comment indicating that every choreo specific database changes should go through this method.